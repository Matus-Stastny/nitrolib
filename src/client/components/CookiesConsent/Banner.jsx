// @flow strict
import * as React from "react";
import styled from "styled-components";

import Text from "client/components/Text";

import { brandDefault } from "client/records/Brand";
import mq from "client/services/utils/mediaQuery";

const Container = styled.div`
  display flex;
  justify-content space-between;
  align-items center;
  box-sizing border-box;
  width: 100%;
  padding 10px 20px;
  font-size 12px;
  background ${({ theme }) => theme.colors["grey-900"]};
  color white;
  opacity .94;
  transition bottom .25s ease-in;
  box-shadow 0 -1px 6px 0 rgba(0, 0, 0, .20);

  ${mq.gtTablet`
    padding 20px;
    color ${({ theme }) => theme.colors["grey-800"]};
    background white;
  `}
`;

Container.defaultProps = {
  theme: brandDefault.theme,
};

const Message = styled.p`
  opacity .5;

  ${mq.gtTablet`
    opacity: 1;
  `};
`;

const InfoLink = styled.a`
  text-decoration underline;
  cursor pointer;

  &:hover {
    color white;
    opacity 1;
  }

  ${mq.gtTablet`
    &:hover {
      color initial;
    }
  `};
`;

const AcceptButton = styled.a`
  padding 10px;
  padding-left 20px;
  font-weight bold;
  white-space nowrap;
  cursor pointer;
`;

type Props = {|
  showInfo: () => void,
  accept: () => void,
|};

const Banner = ({ showInfo, accept }: Props) => (
  <Container>
    <Message>
      <Text t={__("content.cookies.popup.full")} />
      &nbsp;
      <InfoLink onClick={showInfo} data-test="CookiesConsent-ShowInfo">
        <Text t={__("common.learn_more")} />
      </InfoLink>.
    </Message>
    <AcceptButton onClick={accept} data-test="CookiesConsent-Accept">
      <Text t={__("common.got_it")} />
    </AcceptButton>
  </Container>
);

export default Banner;
