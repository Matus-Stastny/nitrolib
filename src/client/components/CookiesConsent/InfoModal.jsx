// @flow strict
import * as React from "react";
import styled from "styled-components";

import Modal from "client/components/Modal";
import Text from "client/components/Text";
import { Consumer as BrandConsumer } from "client/services/brand/context";

const Container = styled.div``;

const Title = styled.h1``;

const Content = styled.p``;

const tKeys = {
  kayakwhite: __("content.cookies.kayakwhite"),
  kiwicom: __("content.cookies.kiwicom"),
  paragraphs: __("content.cookies.paragraphs"),
};

type Props = {|
  onClose: () => void,
|};

const InfoModal = ({ onClose }: Props) => (
  <Modal onClose={onClose}>
    <Container>
      <Title>
        <Text t={__("content.pages.cookies.title")} />
      </Title>
      <Content>
        <BrandConsumer>
          {brand => <Text t={tKeys[brand.content.legal.cookies.t_key]} html />}
        </BrandConsumer>
      </Content>
    </Container>
  </Modal>
);

export default InfoModal;
