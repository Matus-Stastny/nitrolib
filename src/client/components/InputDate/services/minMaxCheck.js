// @flow
import isBefore from "date-fns/isBefore";
import isAfter from "date-fns/isAfter";

export default function minMaxCheck(value: Date, min: Date, max: Date, onChange: Date => void) {
  if (isBefore(value, min)) {
    onChange(min);
  }
  if (isAfter(value, max)) {
    onChange(max);
  }
}
