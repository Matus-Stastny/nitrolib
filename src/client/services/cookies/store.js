// @flow strict
import JsCookies from "js-cookie";

import { COOKIE_CONSENT_KEY, COOKIE_CONSENT_VALUE } from "client/consts/cookies";

export const isAccepted = () => Boolean(JsCookies.get(COOKIE_CONSENT_KEY));

export const saveAccepted = () => {
  JsCookies.set(COOKIE_CONSENT_KEY, COOKIE_CONSENT_VALUE); // TODO: Add cookie options
};
