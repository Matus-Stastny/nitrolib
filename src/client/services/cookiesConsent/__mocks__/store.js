// @flow strict

export const { isAccepted, saveAccepted } = jest.genMockFromModule("../store");

isAccepted.mockReturnValue(false);
