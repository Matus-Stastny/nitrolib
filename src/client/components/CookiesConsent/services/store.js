// @flow strict
import jsCookie from "js-cookie";

import { COOKIE_CONSENT_KEY, COOKIE_CONSENT_VALUE } from "client/consts/cookies";

export const isAccepted = () => Boolean(jsCookie.get(COOKIE_CONSENT_KEY));

export function saveAccepted() {
  jsCookie.set(COOKIE_CONSENT_KEY, COOKIE_CONSENT_VALUE);
}
