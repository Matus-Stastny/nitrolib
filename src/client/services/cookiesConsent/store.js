// @flow strict
import JsCookies from "js-cookie";

import { COOKIE_CONSENT_KEY, COOKIE_CONSENT_VALUE } from "client/consts/cookies";
import { IS_CLIENT } from "client/consts/system";

export const isAccepted = () => !IS_CLIENT || Boolean(JsCookies.get(COOKIE_CONSENT_KEY));

export const saveAccepted = () => {
  if (IS_CLIENT) {
    JsCookies.set(COOKIE_CONSENT_KEY, COOKIE_CONSENT_VALUE);
  }
};
