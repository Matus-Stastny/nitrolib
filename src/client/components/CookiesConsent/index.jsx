// @flow strict
import * as React from "react";

import ClientOnly from "client/components/ClientOnly";
import * as cookiesStore from "client/components/CookiesConsent/services/store";
import Banner from "./Banner";
import InfoModal from "./InfoModal";

type State = {|
  isAccepted: boolean,
  infoShown: boolean,
|};

// TODO: Show consent only in Europe
class CookiesConsent extends React.PureComponent<{}, State> {
  state: State = {
    isAccepted: cookiesStore.isAccepted(),
    infoShown: false,
  };

  handleAccept = () => {
    this.setState({ isAccepted: true });

    cookiesStore.saveAccepted();
  };

  handleShowInfo = () => {
    this.setState({ infoShown: true });
  };

  handleHideInfo = () => {
    this.setState({ infoShown: false });
  };

  render() {
    const { isAccepted, infoShown } = this.state;

    return (
      <ClientOnly>
        {!isAccepted && ( // TODO: Add continent check
          <React.Fragment>
            <Banner onAccept={this.handleAccept} onShowInfo={this.handleShowInfo} />
            {infoShown && <InfoModal onClose={this.handleHideInfo} />}
          </React.Fragment>
        )}
      </ClientOnly>
    );
  }
}

export default CookiesConsent;
