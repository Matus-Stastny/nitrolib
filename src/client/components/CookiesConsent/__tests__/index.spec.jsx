// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import * as store from "../services/store";
import CookiesConsent from "../index";

jest.mock("../services/store");

describe("#CookiesConsent", () => {
  test("render - cookies accepted", () => {
    // $FlowExpected
    store.isAccepted.mockReturnValue(true);

    const wrapper = shallow(<CookiesConsent />);

    expect(store.isAccepted).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });

  test("render - cookies not accepted", () => {
    // $FlowExpected
    store.isAccepted.mockReturnValue(false);

    const wrapper = shallow(<CookiesConsent />);

    expect(store.isAccepted).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });

  test("accept cookies", () => {
    // $FlowExpected
    store.isAccepted.mockReturnValue(false);

    const wrapper = shallow(<CookiesConsent />);

    wrapper.instance().handleAccept();
    wrapper.update();

    expect(store.saveAccepted).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });

  test("showInfo / hideInfo", () => {
    // $FlowExpected
    store.isAccepted.mockReturnValue(false);

    const wrapper = shallow(<CookiesConsent />);

    const instance = wrapper.instance();

    instance.handleShowInfo();
    wrapper.update();

    expect(wrapper).toMatchSnapshot();

    instance.handleHideInfo();
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });
});
