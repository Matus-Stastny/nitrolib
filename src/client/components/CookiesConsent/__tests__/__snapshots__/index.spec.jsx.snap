// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#CookiesConsent accept cookies 1`] = `
<ClientOnly
  loader={null}
/>
`;

exports[`#CookiesConsent render - cookies accepted 1`] = `
<ClientOnly
  loader={null}
/>
`;

exports[`#CookiesConsent render - cookies not accepted 1`] = `
<ClientOnly
  loader={null}
>
  <React.Fragment>
    <Banner
      onAccept={[Function]}
      onShowInfo={[Function]}
    />
  </React.Fragment>
</ClientOnly>
`;

exports[`#CookiesConsent showInfo / hideInfo 1`] = `
<ClientOnly
  loader={null}
>
  <React.Fragment>
    <Banner
      onAccept={[Function]}
      onShowInfo={[Function]}
    />
    <InfoModal
      onClose={[Function]}
    />
  </React.Fragment>
</ClientOnly>
`;

exports[`#CookiesConsent showInfo / hideInfo 2`] = `
<ClientOnly
  loader={null}
>
  <React.Fragment>
    <Banner
      onAccept={[Function]}
      onShowInfo={[Function]}
    />
  </React.Fragment>
</ClientOnly>
`;
